tools/ci_build/github/azure-pipelines/linux-qnn-ci-pipeline.yml

```diff
@@ -63,22 +63,27 @@ jobs:
           python3 tools/ci_build/build.py \
             --build_dir build \
             --config Release \
-            --parallel --use_binskim_compliant_compile_flags \
+            --use_binskim_compliant_compile_flags \
+            --build_java \
             --use_qnn \
             --qnn_home $(QnnSDKRootDir) \
             --cmake_generator=Ninja \
-            --skip_tests
+            --update --build --parallel
         displayName: Build QNN EP
 
       - script: |
+          export ORT_JAVA_TEST_QNN_EP_BACKEND_PATH="$(QnnSDKRootDir)/lib/x86_64-linux-clang/libQnnCpu.so"
+
           python3 tools/ci_build/build.py \
             --build_dir build \
-            --config Release --use_binskim_compliant_compile_flags \
-            --test \
+            --config Release \
+            --use_binskim_compliant_compile_flags \
+            --build_java \
+            --use_qnn \
             --qnn_home $(QnnSDKRootDir) \
             --cmake_generator=Ninja \
             --skip_submodule_sync \
```

Comment:
FWIW some of these flags shouldn't be required when you're just running tests, but maybe it's simpler to keep them to minimize the understanding required.

`--skip_submodule_sync` only applies if `--update` is specified so that's probably the simplest to remove. 

I wouldn't have expected `--use_binskim_compliant_compile_flags` (used in generate_build_tree in build.py but not called if we're only testing) or `--cmake_generator` (we call the test app directly or `cmake test` with the just the build config) to be required either.

Reply:
good point about `--skip_submodule_sync`, removing that. as for the others, I was trying to keep the options unrelated to build phases identical to, as you said, minimize understanding required.